python/across_api/burstcube/api.py

```diff
@@ -1,3 +1,184 @@
+
+    # Open HEALPix fits file and extract data and ordering scheme
+    try:
+        with fits.open(file) as hdu:
+            healpix_data = hdu[1].data
+            healpix_scheme = hdu[1].header["ORDERING"]
+    except (OSError, KeyError, IndexError):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Invalid HEALPix file.",
+        )
+    return healpix_data, healpix_scheme
+
+
+@app.post(
+    "/burstcube/too",
+    status_code=status.HTTP_201_CREATED,
+    dependencies=[
+        Security(scope_authorize, scopes=["gcn.nasa.gov/kafka-public-consumer"])
+    ],
+)
+async def burstcube_too_submit(
+    credential: Annotated[dict, Depends(claims)],
+    ra_dec: OptionalRaDecDep,
+    error_radius: ErrorRadiusDep,
+    trigger_time: TriggerTimeDep,
+    trigger_info: BurstCubeTriggerInfo,
+    exposure: ExposureDep,
+    offset: OffsetDep,
+    healpix_file: UploadFile = File(
+        None, description="HEALPix file describing the localization."
+    ),
+) -> BurstCubeTOOSchema:
+    """
+    Resolve the name of an astronomical object to its coordinates.
+    """
+    # Construct the TOO object.
+    too = BurstCubeTOO(
+        username=credential["sub"],
+        ra=ra_dec["ra"],
+        dec=ra_dec["dec"],
+        error_radius=error_radius,
+        trigger_time=trigger_time,
+        trigger_info=trigger_info,
+        exposure=exposure,
+        offset=offset,
+    )
+    # If a HEALpix file was uploaded, open it and set the healpix_loc
+    # and healpix_scheme attributes.
+    if healpix_file is not None:
+        too.healpix_loc, too.healpix_scheme = read_healpix_file(healpix_file)
+    too.post()
+    return too.schema
+
+
+@app.put(
+    "/burstcube/too/{id}",
+    status_code=status.HTTP_201_CREATED,
+    dependencies=[
+        Security(scope_authorize, scopes=["gcn.nasa.gov/kafka-public-consumer"])
+    ],
+)
+async def burstcube_too_update(
+    id: IdDep,
+    credential: Annotated[dict, Depends(claims)],
+    ra_dec: OptionalRaDecDep,
+    error_radius: ErrorRadiusDep,
+    trigger_time: TriggerTimeDep,
+    trigger_info: BurstCubeTriggerInfo,
+    exposure: ExposureDep,
+    offset: OffsetDep,
+    healpix_file: UploadFile = File(
+        None, description="HEALPix file describing the localization."
+    ),
+) -> BurstCubeTOOSchema:
+    """
+    Update a BurstCube TOO object with the given ID number.
+    """
+    # Update the TOO object.
+    too = BurstCubeTOO(
+        id=id,
+        username=credential["sub"],
+        ra=ra_dec["ra"],
+        dec=ra_dec["dec"],
+        error_radius=error_radius,
+        trigger_time=trigger_time,
+        trigger_info=trigger_info,
+        exposure=exposure,
+        offset=offset,
+    )
+    # If a HEALpix file was uploaded, open it and set the healpix_loc
+    # and healpix_scheme attributes.
+    if healpix_file is not None:
+        too.healpix_loc, too.healpix_scheme = read_healpix_file(healpix_file)
```

Comment:
We should store HEALPix files in S3, not DynamoDB. HEALPix files will exceed the per-item size DynamoDB item size limit and this will fail.

Reply:
Verified, we're not storing healpix map in the database.